docs/book/src/tasks/experimental-features/cluster-class/change-clusterclass.md

```diff
@@ -1,32 +1,192 @@
+- create a new template
+- update the template reference in the ClusterClass
+- then delete the old template
+
+<aside class="note">
+<h1>In place template mutations</h1>
+
+In case a provider supports in place template mutations, the Cluster API topology controller 
+will adapt to them at the next reconciliation, but the system is not watching for those 
+specific changes. That means that when the underlying template is updated in this way the changes 
+may not be reflected immediately, but it will be put in place at the next full reconciliation. 
+The maximum time for the next reconciliation to take place is related to the CAPI controller
+sync period - 10 minutes by default.
+
+</aside>
 
 <aside class="note warning">
+<h1>Reusing templates across ClusterClasses</h1>
+
+As already discussed in [writing a cluster class](write-clusterclass.md), technically it is possible to 
+re-use a template across ClusterClasses, but this practice is not recommended because it makes it more
+complex to reason about the impact that any change on such a template can have on existing Clusters.
+
+</aside>
+
+Also in case of changes to the ClusterClass templates, please make sure to:
+
+- Understand what [Compatibility Checks](#Compatibility Checks) are and how to prevent changes
+  that can lead to non-functional Clusters.
+- [Dry running ClusterClass changes](#Dry running ClusterClass changes) before applying them.
+
+And eventually learn more about this reading the notes in the [dry run](#Dry running ClusterClass changes) documentation or
+looking at the [referece](#Reference) documentation at the end of this page.
+
+## Rebase
 
+Rebasing is an operational practice for transitioning a Cluster from one ClusterClass to another,
+and the operation can be triggered by simply changing the value in `cluster.spec.topology.class`.
+
+Also in this case, please make sure to:
+
+- Understand what [Compatibility Checks](#Compatibility Checks) are and how to prevent changes
+  that can lead to non-functional Clusters.
+- [Dry running ClusterClass changes](#Dry running ClusterClass changes) before applying them.
+
+And eventually learn more about this reading the notes in the [dry run](#Dry running ClusterClass changes) documentation or
+looking at the [referece](#Reference) documentation at the end of this page.
+
+## Compatibility Checks
+
+When changing a ClusterClass, the system validates the required changes according to
+a set of "compatibility rules" trying to prevent changes to leading to a non-functional
+Cluster, like e.g. changing the InfrastructureProvider from AWS to Azure.
+
+If the proposed changes are evaluated as dangerous, the operation is rejected.
+
+<aside class="note warning">
 <h1>Warning</h1>
 
-Changing a ClusterClass triggers changes on all the Clusters using the ClusterClass.
+In the current implementation there are no compatibility rules for provider's
+template changes, so you should refer to the provider documentation for avoiding
```

Comment:
```suggestion
In the current implementation there are no compatibility rules for changes to provider
templates, so you should refer to the provider documentation to avoid
```